src/ucp/proto/proto_am.inl

```diff
@@ -281,6 +281,9 @@ ucp_am_zcopy_single_handle_status(ucp_request_t *req,
         ucp_request_send_state_advance(req, dt_state,
                                        UCP_REQUEST_SEND_PROTO_ZCOPY_AM,
                                        status);
+        if (ucs_unlikely(UCS_STATUS_IS_ERR(status))) {
+            ucp_request_send_state_ff(req, status);
+        }
```

Comment:
maybe move this inside of `ucp_request_send_state_advance`? IMHO it seems weird to change state twice

Reply:
in `state_advance()`, the state just moved to the new one.
in `send_state_ff()`, we move state to the latest one in case of error.

the problem RMA operations that use `ucp_rma_request_advance()`, not all of them need to do `state_advance()`. So, we should separate them

Reply:
It's simpler but not better, is it possible to unify them?

Reply:
> is it possible to unify them?

done